qualang_tools/config/builder.py

```diff
@@ -0,0 +1,211 @@
+            return self._find_users_of_waveform(elm)
+        elif isinstance(elm, Pulse):
+            return self._find_users_of_pulse(elm)
+        elif isinstance(elm, IntegrationWeights):
+            return self._find_users_of_integration_weights(elm)
+        elif isinstance(elm, Mixer):
+            return self._find_users_of_mixer(elm)
+        else:
+            raise NotImplementedError("can not find objects of type {}".format(type(elm)))
+
+    def _find_users_of_port(self, port:Port):
+        objects = []
+        for obj in self.objects:
+            if isinstance(obj, Element) or isinstance(obj, ElementCollection) or isinstance(obj, MeasureElement):
+                if port in obj.ports:
+                    objects.append(obj)
+        return set(objects)
+
+    def _find_users_of_controller(self, cont:Controller):
+        objects = []
+        for obj in self.objects:
+            if isinstance(obj, Element) or isinstance(obj, MeasureElement):
+                for port in obj.ports:
+                    if port in cont.ports:
+                        objects.append(obj)
+                        break
+            elif isinstance(obj, ElementCollection):
+                for _obj in obj.elements:
+                    for port in _obj.ports:
+                        if port in cont.ports:
+                            objects.append(obj)
+                        break
+        return set(objects)
+
+    def _find_users_of_waveform(self, wf:Waveform):
+        objects = []
+        for obj in self.objects:
+            if isinstance(obj, Element) or isinstance(obj, MeasureElement) or isinstance(obj, ElementCollection):
+                for name in obj.waveform_names:
+                    if name == wf.name:
+                        objects.append(obj)
+        return set(objects)
+
+    def _find_users_of_pulse(self, elm:Pulse):
+        objects = []
+        for obj in self.objects:
+            if isinstance(obj, Element) or isinstance(obj, MeasureElement) or isinstance(obj, ElementCollection):
+                for name in obj.pulse_names:
+                    if name == elm.name:
+                        objects.append(obj)
+        return set(objects)
+
+    def _find_users_of_integration_weights(self, elm:IntegrationWeights):
+        objects = []
+        for obj in self.objects:
+            if isinstance(obj, MeasurePulse):
+                for w in obj.integration_weights:
+                    if w.name == elm.name:
+                        objects.append(obj)
+        return set(objects)
+
+    def _find_users_of_mixer(self, elm:Mixer):
+        objects = []
+        for obj in self.objects:
+            if isinstance(obj, Oscillator):
+                if obj.mixer == elm.name:
+                    objects.append(obj)
+            elif isinstance(obj, Element):
+                if obj.mixer.name == elm.name:
+                    objects.append(obj)
+        return set(objects)
+
+    @property
+    def qua_config(self):
```

Comment:
Shouldn't the properties be together?
And isn't this confusing? people will use that instead of build?

Reply:
users should only use `build` method, I removed `qua_config`